src/components/Input/useInput.ts

```diff
@@ -0,0 +1,70 @@
+import { useCallback, useState } from 'react';
+
+import { InputTypeEnum, UseInputProps } from './types';
+
+import { getValidationError } from '../../utils/validation';
+
+export const useInput = ({
+  handleInput,
+  type,
+  value,
+  validationRule,
+}: UseInputProps) => {
+  const [defaultInputValue, defaultInputHandler] = useState('');
+  const [isCrossedEye, setIsCrossedEye] = useState(false);
+  const [errorMessage, setErrorMessage] = useState('');
+  const [currentType, setСurrentType] = useState(type || InputTypeEnum.text);
+
+  const checkValidation = (value: string) => {
+    const { errorMessage } = getValidationError(validationRule, value);
+
+    setErrorMessage(errorMessage);
+  };
+
+  const toggleEye = useCallback(() => {
+    if (isCrossedEye) {
+      setСurrentType(InputTypeEnum.password);
+    } else {
+      setСurrentType(InputTypeEnum.text);
+    }
+
+    setIsCrossedEye((prevState) => !prevState);
+  }, [isCrossedEye]);
+
+  const handleInputChange = (e: React.ChangeEvent<HTMLInputElement>) => {
+    const { value } = e.target;
+
+    checkValidation(value);
+
+    if (handleInput) {
+      handleInput(value);
+    } else {
+      defaultInputHandler(value);
+    }
```

Comment:
🧐
Лучше будет вот так
```suggestion
      handleChange(value);
```

Есть хороший паттерн для того, чтобы избегать подобных конструкций. В момент деструктуризации параметров использовать присваивание
```
const { prop = defaultProp } = props;
```

То есть в момент присвоения положить какую-то функцию в этот деструктурируемый проп (в нашем случае стейт-сеттер) и использовать потом без условий

Reply:
Супер! Спасибо, исправил)